### PR TITLE
llvm10

### PR DESCRIPTION
**Pull request by nsurbay**

### CODE REVIEW

File: src/ExecBlock/ExecBlock.cpp [hunk 1]
```diff
@@ -139,7 +140,7 @@ void ExecBlock::show() const {
         );
 
         llvm::raw_string_ostream disassOs(disass);
-        assembly.printDisasm(inst, disassOs);
+        assembly.printDisasm(inst, reinterpret_cast<uint64_t>(codeBlock.base()) + i, disassOs);
```
Comment (romainthomas) [inline src/ExecBlock/ExecBlock.cpp:143]: Better to use ``uintptr_t`` instead of ``uint64_t``

File: src/Utility/Assembly.cpp [hunk 2]
```diff
@@ -78,9 +78,10 @@ void Assembly::writeInstruction(const llvm::MCInst inst, memory_ostream *stream)
     LogCallback(LogPriority::DEBUG, "Assembly::writeInstruction", [&] (FILE *log) -> void {
         std::string disass;
         llvm::raw_string_ostream disassOs(disass);
-        printDisasm(inst, disassOs);
+        uint64_t address = (uint64_t) stream->get_ptr() + (uint64_t) pos;
```
Comment (romainthomas) [inline src/Utility/Assembly.cpp:81]: ``uintptr_t`` instead of of ``uint64_t``

File: src/Utility/Assembly.cpp [hunk 3]
```diff
@@ -107,9 +108,9 @@ void Assembly::writeInstruction(const llvm::MCInst inst, memory_ostream *stream)
 }
 
 
-void Assembly::printDisasm(const llvm::MCInst &inst, llvm::raw_ostream &out) const {
+void Assembly::printDisasm(const llvm::MCInst &inst, uint64_t address, llvm::raw_ostream &out) const {
```
Comment (romainthomas) [inline src/Utility/Assembly.cpp:111]: ``uintptr_t`` instead of of ``uint64_t``

File: src/Utility/Assembly.h [hunk 4]
```diff
@@ -61,7 +61,7 @@ class Assembly {
     llvm::MCDisassembler::DecodeStatus getInstruction(llvm::MCInst &inst, uint64_t &size,
                                             llvm::ArrayRef<uint8_t> bytes, uint64_t address) const;
 
-    void printDisasm(const llvm::MCInst &inst, llvm::raw_ostream &out = llvm::errs()) const;
+    void printDisasm(const llvm::MCInst &inst, uint64_t address, llvm::raw_ostream &out = llvm::errs()) const;
```
Comment (romainthomas) [inline src/Utility/Assembly.h:64]: ``uintptr_t`` instead of of ``uint64_t``

File: test/TestSetup/InMemoryAssembler.cpp [hunk 5]
```diff
@@ -129,12 +130,16 @@ InMemoryObject::InMemoryObject(const char* source, const char* cpu, const char**
...
-            if((ec = sit->getContents(text_section))) {
-                llvm::errs() << "Failed to load text section: " << ec.message() << "\n";
-                return;
-            }
-            else {
+            if (llvm::Expected<llvm::StringRef> E  = sit->getContents()) {
+                text_section = *E;
                 break;
+            } else {
+                llvm::handleAllErrors(E.takeError(), [&](const llvm::ErrorInfoBase &EIB) {
```
Comment (romainthomas) [inline test/TestSetup/InMemoryAssembler.cpp:137]: Why do you need to capture with ``[&]`` ?

File: src/ExecBlock/ExecBlock.cpp (hunk 1 above)
Comment (nsurbay) [reply src/ExecBlock/ExecBlock.cpp:143]: I prefer to keep a uint64_t to match the LLVM API (https://github.com/llvm/llvm-project/blob/llvmorg-10.0.1/llvm/include/llvm/MC/MCDisassembler/MCDisassembler.h#L77)

File: test/TestSetup/InMemoryAssembler.cpp (hunk 5 above)
Comment (nsurbay) [reply test/TestSetup/InMemoryAssembler.cpp:137]: No need or the ``[&]``. I copy some llvm code that handle error and forget to remove the capture.

File: src/Utility/Assembly.cpp (hunk 3 above)
Comment (nsurbay) [reply src/Utility/Assembly.cpp:111]: I prefer to keep a ``uint64_t`` to match the LLVM API (https://github.com/llvm/llvm-project/blob/llvmorg-10.0.1/llvm/include/llvm/MC/MCDisassembler/MCDisassembler.h#L77)

File: src/Utility/Assembly.h (hunk 4 above)
Comment (nsurbay) [reply src/Utility/Assembly.h:64]: I prefer to keep a ``uint64_t`` to match the LLVM API (https://github.com/llvm/llvm-project/blob/llvmorg-10.0.1/llvm/include/llvm/MC/MCDisassembler/MCDisassembler.h#L77)

File: src/Utility/Assembly.cpp (hunk 2 above)
Comment (nsurbay) [reply src/Utility/Assembly.cpp:81]: see previous comment.
I change the C cast to an reinterpret_cast.

**Pull request merged**